llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -1466,12 +1465,9 @@ class LoopVectorizationCostModel {
     // Override forced styles if needed.
     // FIXME: use actual opcode/data type for analysis here.
     // FIXME: Investigate opportunity for fixed vector factor.
-    bool EVLIsLegal =
-        IsScalableVF && UserIC <= 1 &&
```

Comment:
Is dropping `IsScalable` intentional here? Is it related to reasoning about the safe distance?

Reply:
Yes. With this change, we do not know yet here, if VF is scalable or not, so just set the tail folding, choose VF, and then disable tail folding, if for some reason it does not work.